src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilderCache.cs

```diff
@@ -70,22 +67,43 @@ public ContentItemCollection GetContentItems(LockFileLibrary library, LocalPacka
                     collection.Load(library.Files);
                 }
 
-                _contentItems.Add(identity, collection);
-            }
+                return collection;
+            });
+        }
 
-            return collection;
+        /// <summary>
+        /// Try to get a LockFileTargetLibrary from the cache.
+        /// </summary>
+        internal LockFileTargetLibrary GetLockFileTargetLibrary(RestoreTargetGraph graph, NuGetFramework framework, LocalPackageInfo localPackageInfo, string aliases, LibraryIncludeFlags libraryIncludeFlags, Func<LockFileTargetLibrary> valueFactory)
+        {
+            localPackageInfo = localPackageInfo ?? throw new ArgumentNullException(nameof(localPackageInfo));
+            var criteriaKey = new CriteriaKey(graph.TargetGraphName, framework);
+            var package = (localPackageInfo.Id, localPackageInfo.Version, localPackageInfo.Sha512);
```

Comment:
Good point. I considered it but I was afraid that path comparison isn't trivial to do right (e.g. case sensitive vs. case insensitive filesystems or different directory separator path). 

If we want to keep it simple and we can just assume case sensitive file systems and assume that paths are canonical.
I'll see whether it is any faster comparing to the current value tuple or not.

Reply:
Look into PathUtility.GetStringComparerBasedOnOS().

Reply:
@nkolev92 One more PR for using a package path instead of a tuple https://github.com/marcin-krystianc/NuGet.Client/pull/2

I think that we don't have to worry whether a file system is case sensitive or not for several reasons:
- It is not easy to implement custom comparer for complex value tuples - it is better to avoid it if not is not really necessary
- `LocalPackageInfo` is cached so we will get consistent values each time for the same package identity anyway
- `VersionFolderPathResolver` normalizes paths to lower case (See https://github.com/NuGet/NuGet.Client/blob/dffef7f001cc49233ed6c4af0a417897b3280bea/src/NuGet.Core/NuGet.Packaging/VersionFolderPathResolver.cs#L29)

Reply:
@nkolev92 Can you have a look at my comment above ?

Reply:
Done. 
Sorry for the delay. 
Please make sure to rebase as well.

Reply:
No problem, done.